src/main/scala/uk/gov/nationalarchives/tdr/api/service/FFIDMetadataService.scala

```diff
@@ -93,12 +86,18 @@ class FFIDMetadataService(ffidMetadataRepository: FFIDMetadataRepository,
     } yield rows
   }
 
-  def checkStatus(puid: Option[String], consignmentType: String): String = {
-    puid.getOrElse("") match {
-      case p if passwordProtectedPuids.contains(p) => PasswordProtected
-      case p if zipPuids.contains(p) => Zip
-      case p if consignmentType == ConsignmentType.judgment && !judgmentPuidsAllow.contains(p) => NonJudgmentFormat
-      case _ => Success
+  def checkStatus(puidOption: Option[String], consignmentType: String): Future[String] = {
```

Comment:
I would split this function out into separate ones, that way you're not making an unnecessary call to the repo. It would also make the logic a bit clearer. Something like this:
```
...
def checkStatus(puid: String, consignmentType: String): Future[String] = {
    if (consignmentType == ConsignmentType.judgment) {
     checkJudgmentStatus(puid)
    }  else {
     checkStandardStatus(puid)
    }
  }

  def checkJudgmentStatus(puid: String): Future[String] = {
    for {
      allowedPuidExists <- puidRepository.checkAllowedPuidExists(puid)
    } yield if (allowedPuidExists) { Success } else { NonJudgmentFormat }
  }

  def checkStandardStatus(puid: String): Future[String] = {
    for {
      disallowedPuidReason <- puidRepository.getDisallowedPuidReason(puid).map(_.getOrElse(""))
    } yield disallowedPuidReason match {
      case reason if reason == PasswordProtected || reason == Zip => reason
      case _ => Success
    }
...
```
It will also allow you to unit test the individual logic in each of the new functions

Reply:
I have split out the logic as suggested. However `checkJudgmentStatus` will still need to call the disallowed table for cases where a user uploads a `PasswordProtected` document.